packages/shell-api/bin/api-postprocess.ts

```diff
@@ -0,0 +1,156 @@
+            this.processedMethods.find(
+              ([cls, method, classBody]) =>
+                cls === className &&
+                method === methodName &&
+                classBody !== path.parent
+            )
+          ) {
+            throw new Error(`Duplicate method: ${className}.${methodName}`);
+          }
+          this.processedMethods.push([className, methodName, path.parent]);
+
+          const classHelp = (enUs['shell-api'] as any).classes[className];
+          if (
+            classHelp &&
+            classHelp.help.attributes &&
+            classHelp.help.attributes[methodName]
+          ) {
+            const methodHelp = classHelp.help.attributes[methodName];
+            if (methodHelp && methodHelp.description) {
+              //console.log(`${className}.${methodName}`, methodHelp.description);
+              path.addComment(
+                'leading',
+                `\n${methodHelp.description as string}\n`,
+                false
+              );
+            }
+          }
+
+          if (!signatures[className]?.attributes?.[methodName]?.returnsPromise)
+            return;
+
+          const { returnType } = path.node;
+          if (returnType?.type !== 'TSTypeAnnotation') return;
+          if (returnType.typeAnnotation.type !== 'TSTypeReference') return;
+          if (returnType.typeAnnotation.typeName.type !== 'Identifier') return;
+          if (returnType.typeAnnotation.typeName.name !== 'Promise') return;
+          if (!returnType.typeAnnotation.typeParameters?.params.length) return;
+          path.replaceWith({
+            ...path.node,
+            returnType: {
+              ...returnType,
+              typeAnnotation:
+                returnType.typeAnnotation.typeParameters.params[0],
+            },
+            isMongoshAsyncRewrittenMethod: true,
+          });
+        },
+      },
+    };
+  };
+}
+
+async function main() {
+  const apiRaw = await fs.readFile(
+    path.resolve(__dirname, '..', 'lib', 'api-raw.d.ts'),
+    'utf8'
+  );
+  const result = babel.transformSync(apiRaw, {
+    code: true,
+    ast: false,
+    configFile: false,
+    babelrc: false,
+    browserslistConfigFile: false,
+    compact: false,
+    sourceType: 'module',
+    plugins: [applyAsyncRewriterChanges()],
+    parserOpts: {
+      plugins: ['typescript'],
+    },
+  });
+  const apiGlobals = await fs.readFile(
+    path.resolve(__dirname, '..', 'src', 'api-globals.d.ts'),
+    'utf8'
+  );
+
+  const code = (result?.code ?? '') + '\n' + apiGlobals;
+  await fs.writeFile(
+    path.resolve(__dirname, '..', 'lib', 'api-processed.d.ts'),
+    code
+  );
+
+  const exportCode = `"use strict";
+module.exports = ${JSON.stringify(code)};
+`;
+  await fs.writeFile(
+    path.resolve(__dirname, '..', 'lib', 'api-export.js'),
+    exportCode
+  );
+
+  await fs.writeFile(
+    path.resolve(__dirname, '..', 'lib', 'api-export.d.ts'),
+    'declare const _default: string; export default _default;'
+  );
```

Comment:
Just gonna say, my gut feeling is that it would be easiest to just avoid default exports (good practice anyway) and stick to a named export à la

```suggestion
  const exportCode = `"use strict";
module.exports = { api: ${JSON.stringify(code)} };
`;
  await fs.writeFile(
    path.resolve(__dirname, '..', 'lib', 'api-export.js'),
    exportCode
  );

  await fs.writeFile(
    path.resolve(__dirname, '..', 'lib', 'api-export.d.ts'),
    'export declare const api;'
  );
```

instead of jumping through hooks to get a default export to work (definitely non-blocking though)

Reply:
Sure I don't mind. It isn't exactly a public API anyway 😆